server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -668,4 +668,27 @@ public Object externalValue() {
      * Get dynamic runtime fields created while parsing.
      */
     public abstract List<RuntimeFieldType> getDynamicRuntimeFields();
+
+    /**
+     * Find a dynamic mapping template for the given field and its matching type
+     *
+     * @param fieldName the name of the field
+     * @param matchType the expecting matchType of the field
+     * @return the matching template; otherwise returns null
+     * @throws MapperParsingException if the given field has a dynamic template name hint, but no template matches that name hint.
+     */
+    public final DynamicTemplate findDynamicTemplate(String fieldName, DynamicTemplate.XContentFieldType matchType) {
+        final String pathAsString = path().pathAsText(fieldName);
+        final String matchTemplateName = sourceToParse().dynamicTemplateHints().get(pathAsString);
+        for (DynamicTemplate template : root().dynamicTemplates()) {
+            if (template.match(matchTemplateName, pathAsString, fieldName, matchType)) {
+                return template;
+            }
+        }
+        if (matchTemplateName != null) {
+            throw new MapperParsingException(
+                "Can't find dynamic template for dynamic template name hint [" + matchTemplateName + "] of field [" + pathAsString + "]");
```

Comment:
this is what I think makes it not a hint, that I was referring to above in another comment